Modernize development build & dependencies

Dewb opened the pull request:
* Update for node 12, set `engine` property in package.json
* Fix `npm audit` issues
* Run `npx uneject` to restore create-react-app webpack config, reduce direct dependencies
* Remove other unused dependencies
* Fix security warnings in Chrome console
* Fix mixed-standard module exports

mike-d-adsk committed:
Update to Node 12; fix npm audit issues; fix other security warnings; uneject create-react-app; remove unused dependencies

mike-d-adsk committed:
Update README.md

mjkkirschner commented on README.md:
```diff
@@ -1,35 +1,22 @@
-
-## Deploy to Autodesk (Staging)
-
-- Confirm that `homepage` in `package.json` is set to `http://staging-dictionary.dynamobim.com/`. The developer may change this address depending on the staging environment.
```
were these docs moved somewhere internal? I think @zeusongit just documented these here.

mmisol replied:
Usually it's not a good idea to make staging environments public, so the change makes sense.

mjkkirschner commented on src/util/resort.js:
```diff
@@ -47,10 +47,6 @@ resort.objectify = function (ad, q) {
         var ml = [];
         ad.forEach(function (d, i) {
             var c = d.Categories;
-            var parent = "Home";
```
could you comment on this change?

Dewb replied:
This eliminates an unused variable warning in the development server console.

mmisol commented on src/components/Home.js:
```diff
@@ -5,7 +5,11 @@ function Home(props){
+          <a href="http://dynamoprimer.com/" target="_blank" rel="noopener noreferrer" style={{'color':'orangered'}}>Dynamo Primer</a>, 
+          this dictionary is open-source - check it out on our 
+          <a href="https://github.com/DynamoDS/DynamoDictionary" target="_blank" rel="noopener noreferrer" style={{'color':'orangered'}} >Github page</a> 
+          and contribute!</p>
```
For this part, the text shows next to the link, without spaces separating them:
<img width="716" alt="Screen Shot 2020-08-21 at 8 58 06 AM" src="https://user-images.githubusercontent.com/10048120/90893165-78cf4e80-e38c-11ea-92ad-91f21ea8ea32.png">

mmisol reviewed (commented):
Only one minor text issue related to changes here.
Other that that:

- The "Create Pull Request" functionality hangs for me, but I'm not sure if this ever worked when running local.

- Another thing I noticed is editing the description of a node does not show up as a change in the "Create Pull Request" dialog. Should it?

- The unit test is still failing, but it was failing already, so no change there I guess.

mmisol committed:
Fix spacing issue and test

Fixes a spacing issue between the text and the links in the home page.

Also fixes the only unit test, by creating the App component in the
same way it's created by the actual application.

QilongTang commented:
@Dewb Back to this PR late, would you explain Run npx uneject to restore create-react-app webpack config, reduce direct dependencies? Are we still using webpack here? I am looking at a legacy PR https://github.com/DynamoDS/DynamoDictionary/pull/22 which bring in another webpack package [html-loader](https://webpack.js.org/loaders/html-loader/) as dependency and trying to figure out the conflict after this PR and merge that in. Appreciate that.

`npx uneject` failed for me locally
